rmw_fastrtps_shared_cpp/include/rmw_fastrtps_shared_cpp/custom_participant_info.hpp

```diff
@@ -219,12 +220,17 @@ class ParticipantListener : public eprosima::fastdds::dds::DomainParticipantList
         rmw_qos_profile_t qos_profile = rmw_qos_profile_unknown;
         rtps_qos_to_rmw_qos(proxyData.m_qos, &qos_profile);
 
+        const auto & userDataValue = proxyData.m_qos.m_userData.getValue();
+        const auto type_hash = rmw_dds_common::parse_type_hash_from_user_data(
```

Comment:
This needs to be updated with the new API from `rmw_dds_common`, and also needs to do error checking.

Reply:
For the type checking, in all three rmw impls - I'm doing "if not OK then hash = 0", sound reasonable?

Reply:
That seems reasonable.